src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1541,1851 @@ interface MPSMatrixMultiplication {
+		[DesignatedInitializer]
+		IntPtr Constructor (MPSMatrix[] sourceMatrices, MPSMatrix[] destinationMatrices, [NullAllowed] MPSVector offsets, nuint byteOffset);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSKernel))]
+	interface MPSMatrixCopy {
+		[Export ("initWithDevice:copyRows:copyColumns:sourcesAreTransposed:destinationsAreTransposed:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, nuint copyRows, nuint copyColumns, bool sourcesAreTransposed, bool destinationsAreTransposed);
+
+		[Export ("copyRows")]
+		nuint CopyRows { get; }
+
+		[Export ("copyColumns")]
+		nuint CopyColumns { get; }
+
+		[Export ("sourcesAreTransposed")]
+		bool SourcesAreTransposed { get; }
+
+		[Export ("destinationsAreTransposed")]
+		bool DestinationsAreTransposed { get; }
+
+		[Export ("encodeToCommandBuffer:copyDescriptor:")]
+		void Encode (IMTLCommandBuffer cmdBuf, MPSMatrixCopyDescriptor copyDescriptor);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSKernel))]
+	interface MPSImageCopyToMatrix {
+		[Export ("destinationMatrixOrigin", ArgumentSemantic.Assign)]
+		MTLOrigin DestinationMatrixOrigin { get; set; }
+
+		[Export ("destinationMatrixBatchIndex")]
+		nuint DestinationMatrixBatchIndex { get; set; }
+
+		[Export ("dataLayout")]
+		MPSDataLayout DataLayout { get; }
+
+		[Export ("initWithDevice:dataLayout:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, MPSDataLayout dataLayout);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+
+		[Export ("encodeToCommandBuffer:sourceImage:destinationMatrix:")]
+		void Encode (IMTLCommandBuffer commandBuffer, MPSImage sourceImage, MPSMatrix destinationMatrix);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSKernel))]
+	interface MPSImageFindKeypoints {
+		[Export ("keypointRangeInfo")]
+		MPSImageKeypointRangeInfo KeypointRangeInfo { get; }
+
+		[Export ("initWithDevice:info:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, MPSImageKeypointRangeInfo info);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+
+		[Export ("encodeToCommandBuffer:sourceTexture:regions:numberOfRegions:keypointCountBuffer:keypointCountBufferOffset:keypointDataBuffer:keypointDataBufferOffset:")]
+		void Encode (IMTLCommandBuffer commandBuffer, IMTLTexture source, MTLRegion regions, nuint numberOfRegions, IMTLBuffer keypointCountBuffer, nuint keypointCountBufferOffset, IMTLBuffer keypointDataBuffer, nuint keypointDataBufferOffset);
+	}
+	
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSBinaryImageKernel))]
```

Comment:
DisableDefaultCtor ?
Not fully sure in this case since there are setters (it's mutable) so I guess it depends if this can be initialized without a device (if not then `init` is unusable). Please check base type and, if `init` makes sense, add a comment

Reply:
Tested in Xcode, looks like you're right.  
```
/BuildRoot/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MetalPerformanceShaders-88.2/MPSImage/Filters/MPSImageArithmetic.mm:463: failed assertion `Cannot directly initialize MPSImageArithmetic. Use one of the sub-classes of MPSImageArithmetic.
```